picard/ui/matchdetailsdialog/matchdetails.py

```diff
@@ -0,0 +1,211 @@
+            ('rec_id',               "_Recording ID"),
+            ('rec_title',            "Recording Title"),
+            ('rec_len',              "Recording\nLength"),
+            ('rec_artist',           "Recording\nArtist"),
+            ('rg_id',                "_Release Group ID"),
+            ('rg_title',             "Release Group\nTitle"),
+            ('rg_pri_type',          "Release Group\nPri Type"),
+            ('rg_sec_type',          "Release Group\nSec Types"),
+            ('rel_id',               "_Release ID"),
+            ('rel_ctry',             "Release\nCountry"),
+            ('rel_dt',               "Release\nDate"),
+            ('med_fmt',              "Medium\nFormat"),
+            ('med_pos',              "Medium\nPosition"),
+            ('med_tot_cnt',          "Medium\nCount"),
+            ('trk_id',               "_Track ID"),
+            ('trk_pos',              "Track\nPosition"),
+            ('med_trk_cnt',          "Medium\nTrack\nCount"),
+            ('rel_tot_trk_cnt',      "Release\nTotal Track\nCount"),
+            ('parts',                "_Scoring Parts"),
+            ('release_parts',        "_Scoring Release Parts"),
+            ('recording_base_score', "_Recording\nBase Score\n(Sources Ratio)"),
+            ('acoust_id',            "AcoustId ID"),
+            ('rec_src_cnt',          "AcoustId\nRecording\nSource\nCount"),
+            ('similarity',           "Similarity")
+        ]
+
+    def load_match_details(self, track, file_):
+        """ Retrieves data and loads it to the screen/dialog box """
+        if file_ is not None:
+            # NOTE: some of the values, like meta, are not exposed as constants
+            fp = file_.acoustid_fingerprint if hasattr(file_, 'acoustid_fingerprint') else ""
+            dur = file_.acoustid_length if hasattr(file_, 'acoustid_length') else ""
+
+            fingerprint_lkp_url = ("https://" + ACOUSTID_HOST + "/v2/lookup"
+                                   + "?meta=recordings%20releasegroups%20releases%20tracks%20compress%20sources"
+                                   + "&fingerprint=" + fp
+                                   + "&duration=" + str(dur)
+                                   + "&client=" + ACOUSTID_KEY
+                                   + "&clientversion=" + PICARD_VERSION_STR
+                                   + "&format=json") if fp is not None else ""
+
+            selected_entry_nbr = match_details_log.get_file_result_selected_entry_nbr(file_.filename)
+            self.set_input_filename_lbl(file_.filename)
+            self.set_acoustid_lookup_url_edit(fingerprint_lkp_url)
+
+        self.current_file = file_
+        self.prepare_table()
+        self.entries = match_details_log.get_match_dtls(file_.filename)
+        display_row_nbr = -1
+        selected_row_nbr = -1
+        for entry_nbr, entry in enumerate(self.entries):
+            if entry.match_dtl_entry_type == "Recording candidate release score":
```

Comment:
Perhaps we should use a numeric type rather than the label directly.
It has various major advantages:
- if you want to change the text, you change it at one place
- it speeds up this comparaison
- it shows clearly what are all possible types
- those numeric types can be whatever you want (class attributes, module level "constants", etc.)

The overall idea is something like:

```python

class MatchDetailsEntryType:
    yourtype1 = 1
    yourtype2 = 2
    ...

type2label = {
    MatchDetailsEntryType.yourtype1: N_('your label'),
     ...
}

if entry.match_dtl_entry_type == MatchDetailsEntryType.yourtype1:
    label = type2label[MatchDetailsEntryType.yourtype1]
    # you may mark the string for translation in the type2label with N_() and translate with _(label)
    do_something_with_label(_(label))
    ...


match_details_log.MatchDtlEntry(MatchDetailsEntryType.yourtype1, ...)
```

Reply:
Done.  Yes you are absolutely right.  I had previously considered making it an enum, and now I've made the change.